### DISCUSSION TITLE
Add a patches directory with similar structure as ungoogled-chromium/patches

### DISCUSSION THREAD

**Discussion by BugmanSpirit** [Alex313031/thorium#163] [Ideas]
Hi,
It would be nice if there was a patches folder in the repository with similar folder structure as https://github.com/ungoogled-software/ungoogled-chromium/tree/master/patches. As mentioned in the wiki, the patches that were borrowed from ungoogled-chromium and other projects could be placed in the folder with that name and the patches created by thorium could be placed in the folder patches/thorium/. All patches for compiler optimization flags, privacy improvements and other enhancements except the branding changes would be helpful for other similar chromium projects to use and contribute.

It would also be helpful for thorium users to add patches from other projects if they want to further harden the privacy and degoogle their browser if they wish to.

## Use GitHub Actions for builds
It would be really easy for individual users to just fork the thorium repo, add a few more patches and build/test their version of hardened thorium if there was a github actions workflow which would add all the patches from the patches folder before building. It would also mean that users don't have to have a development setup powerful enough t do a chromium build. I think github actions is free for open source projects and it would also save the developers some resources.

**Comment by gz83** [comment]
This markdown file lists the patches used by Thorium and has indicated the source

https://github.com/Alex313031/Thorium/blob/main/infra/PATCHES.md

Automated tools such as Github actions are not suitable for building large-scale projects at this stage. They have certain limitations in usage time and resource occupation. If you use automated tools such as Github actions to build, it will greatly increase the compilation time of Thorium. There is a comparison a good solution is to use a distributed compilation tool like GOMA.

@BugmanSpirit

**Comment by Alex313031** [comment]
@BugmanSpirit As @gz83 already knows, I prefer not using .patch files, and instead have the full source laid out in a directory tree mirroring the chromium one. For the libjxl repo, we have .patch files that can be applied to vanilla chromium. For thorium, if you want info about a specific patch + the source of it, for example to apply only a specific patch, then you can use the PATCHES.md file that gz83 linked above.

Thorium uses patches from many projects, and also some patches that we have created ourselves. I add stuff sparingly, and only add things if a large portion of the community wants it, or it is something most people would want enabled. As per the readme, one of thorium's goals is to enhance privacy and security, but NOT at the expense of usability. This is why some patches that people have suggested from brave or ungoogled have not been added, as they interfere with usability  (oftentimes google integration). While google is not a "nice guy" the reality is that most people still want to use it. For the tin foil hatters, ungoogled chromium is the best choice if you want complete autonomy from Google. You will miss out on thoriums performance though lol.
